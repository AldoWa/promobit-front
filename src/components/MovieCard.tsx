import { GridItem, Image, Skeleton, Text } from '@chakra-ui/react';
import React from 'react';
import { Link } from 'react-router-dom';

interface MovieProps {
  src: string;
  title: string;
  year: string;
  id: string;
}

const MovieCard: React.FC<MovieProps> = ({ title, year, src, id }) => {
  return (
    <GridItem width={176} alignItems='start' justifyContent='start' >
      <Link to={`/details/${id}`}>
        <Image
          src={src}
          alt={`Movie ${title}`}
          width='100%'
          height={264}
          borderRadius={4}
          fallback={<Skeleton width={176} height={264} borderRadius={4} />}
          data-testid="movieCard-image"
          >
        </Image>
        <Text color='black' fontSize='md' textAlign='start' fontWeight={700} mt="8px" data-testid="movieCard-title">{title}</Text>
        <Text color='#646464'
          fontSize='sm'
          textAlign='start'
          fontWeight={700}
          width='100%'
          mt="4px"
          data-testid="movieCard-year"
        >{year}</Text>
      </Link>
    </GridItem>
  );
};

export default MovieCard;
