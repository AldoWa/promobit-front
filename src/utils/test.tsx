import { RenderOptions, render } from '@testing-library/react'

import { ChakraProvider } from '@chakra-ui/react'

interface AllTheProvidersProps {
  children: React.ReactNode
}

const AllTheProviders = ({children}: AllTheProvidersProps) => {
  return (
    <ChakraProvider>
      {children}
    </ChakraProvider>
  )
}

const customRender = (ui:React.ReactElement, options?: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// eslint-disable-next-line react-refresh/only-export-components
export * from '@testing-library/react'

// override render method
export {customRender as render}