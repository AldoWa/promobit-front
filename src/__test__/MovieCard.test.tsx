import { render, screen } from '../utils/test';
import MovieCard from '../components/MovieCard';

describe('MovieCard', () => {
  it('Should render the MovieCard with the props', async () => {
    render(<MovieCard id='1' src='aletory' title='Mortal Kombat' year='2021'/>)
    
    const titleMovie = screen.getByTestId('movieCard-title')
    const yearMovie = screen.getByTestId('movieCard-year')

    expect(titleMovie).toBeInTheDocument()
    expect(titleMovie).toHaveTextContent('Mortal Kombat')

    expect(yearMovie).toBeInTheDocument()
    expect(yearMovie).toHaveTextContent('2021')
  });

  it("Should match snapshot", () => {
    const { container } = render(<MovieCard id='1' src='aletory' title='Mortal Kombat' year='2021' />)
    expect(container).toMatchSnapshot();
  })
})