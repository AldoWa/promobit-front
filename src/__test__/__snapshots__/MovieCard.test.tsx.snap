// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard Should match snapshot 1`] = `
<div>
  <div
    class="css-9jm09p"
  >
    <a
      href="/details/1"
    >
      <div
        class="chakra-skeleton css-218kxs"
      />
      <p
        class="chakra-text css-1ara00f"
        data-testid="movieCard-title"
      >
        Mortal Kombat
      </p>
      <p
        class="chakra-text css-1hqom77"
        data-testid="movieCard-year"
      >
        2021
      </p>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
